src/m365/spo/commands/file/file-roleassignment-remove.spec.ts

```diff
@@ -0,0 +1,262 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import appInsights from '../../../../appInsights';
+import auth from '../../../../Auth';
+import { Cli } from '../../../../cli/Cli';
+import { CommandInfo } from '../../../../cli/CommandInfo';
+import { Logger } from '../../../../cli/Logger';
+import Command, { CommandError } from '../../../../Command';
+import request from '../../../../request';
+import { urlUtil } from '../../../../utils/urlUtil';
+import { sinonUtil } from '../../../../utils/sinonUtil';
+import commands from '../../commands';
+import * as SpoFileGetCommand from './file-get';
+import * as SpoGroupGetCommand from '../group/group-get';
+import * as SpoUserGetCommand from '../user/user-get';
+const command: Command = require('./file-roleassignment-remove');
+
+describe(commands.FILE_ROLEASSIGNMENT_REMOVE, () => {
+  const webUrl = 'https://contoso.sharepoint.com/sites/contoso-sales';
+  const fileUrl = '/sites/contoso-sales/documents/Test1.docx';
+  const fileId = 'b2307a39-e878-458b-bc90-03bc578531d6';
+  const principalId = 2;
+  const upn = 'user1@contoso.onmicrosoft.com';
+  const groupName = 'saleGroup';
+
+  let log: any[];
+  let logger: Logger;
+  let commandInfo: CommandInfo;
+  let promptOptions: any;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    sinon.stub(Cli, 'prompt').callsFake(async (options: any) => {
+      promptOptions = options;
+      return { continue: false };
+    });
+    promptOptions = undefined;
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      Cli.prompt,
+      request.post
+    ]);
+  });
```

Comment:
```suggestion
  afterEach(() => {
    sinonUtil.restore([
      Cli.prompt,
      Cli.executeCommandWithOutput,
      request.post
    ]);
  });
```

@nanddeepn I think that you missed this one single line which will resolve the issue you are having with the test 😉.
I don't have the possibility to double check it now (only quick check over the phone). I will try to do it ASAP, but not sure that I will manage today 🙂, but please start with resetting the `CLI.executeCommandWithOutput` after each test.

have an awesome rest of the week 👍. You rock 🤩

Reply:
Think this will indeed fix it!

You can only stub something once. Before you stub it again, you should clear the previous stub. The sinonUtil makes sure all stubs are cleared for the provided functions.

Reply:
Thank you very much @Adam-it @milanholemans 
You catched it correctly. You are awesome 🤩

Reply:
thanks 🙂. Now I do feel awesome 🙂
![IFeelAwesomeEricCartmanGIF](https://user-images.githubusercontent.com/58668583/193942505-39dbd077-8658-4a73-91d5-8a7c378efa3c.gif)